plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -221,6 +237,99 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
+                    }
+                    JdbcTypeHandle typeHandle = new JdbcTypeHandle(
+                            getInteger(resultSet, "DATA_TYPE").orElseThrow(() -> new IllegalStateException("DATA_TYPE is null")),
+                            typeName,
+                            getInteger(resultSet, "COLUMN_SIZE"),
+                            decimalDigits,
+                            Optional.empty(),
+                            Optional.empty());
+                    Optional<ColumnMapping> columnMapping = toColumnMapping(session, connection, typeHandle);
+                    log.debug("Mapping data type of '%s' column '%s': %s mapped to %s", schemaTableName, columnName, typeHandle, columnMapping);
+                    // skip unsupported column types
+                    boolean nullable = (resultSet.getInt("NULLABLE") != columnNoNulls);
+                    // Note: some databases (e.g. SQL Server) do not return column remarks/comment here.
+                    Optional<String> comment = Optional.ofNullable(emptyToNull(resultSet.getString("REMARKS")));
+                    if (columnMapping.isPresent()) {
+                        columns.add(JdbcColumnHandle.builder()
+                                .setColumnName(columnName)
+                                .setJdbcTypeHandle(typeHandle)
+                                .setColumnType(columnMapping.get().getType())
+                                .setNullable(nullable)
+                                .setComment(comment)
+                                .build());
+                    }
+                    if (columnMapping.isEmpty()) {
+                        UnsupportedTypeHandling unsupportedTypeHandling = getUnsupportedTypeHandling(session);
+                        verify(
+                                unsupportedTypeHandling == IGNORE,
+                                "Unsupported type handling is set to %s, but toTrinoType() returned empty for %s",
+                                unsupportedTypeHandling,
+                                typeHandle);
+                    }
+                }
+                if (columns.isEmpty()) {
+                    // A table may have no supported columns. In rare cases (e.g. PostgreSQL) a table might have no columns at all.
+                    throw new TableNotFoundException(
+                            schemaTableName,
+                            format("Table '%s' has no supported columns (all %s columns are not supported)", schemaTableName, allColumns));
+                }
+                return ImmutableList.copyOf(columns);
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+    }
+
+    private static Map<String, Integer> getTimestampPrecisions(Connection connection, JdbcTableHandle tableHandle)
+            throws SQLException
+    {
+        String sql = "" +
+                "SELECT column_name, datetime_precision " +
```

Comment:
Is this column available in all supported versions (>= 5.7)?

Reply:
We should have tests exercising all supported versions, so we don't have to ask  this ourselves
(would connector smoke test coverage be enough?)